include/SugarObjects/templates/basic/Basic.php

```diff
@@ -134,24 +179,23 @@ public function getEmailAddressConfirmOptIn($emailField)
      * @throws RuntimeException
      * @throws InvalidArgumentException
      */
-    private function getEmailAddressId($emailField)
+    protected function getEmailAddressId($emailField)
     {
-        $this->getEmailAddressValidateArguments($emailField);
+        global $log;
 
         $emailAddress = $this->cleanUpEmailAddress($this->{$emailField});
 
         if (!$emailAddress) {
-            global $log;
+
             $log->warn('Trying to get an empty email address.');
             return null;
         }
 
-        if (empty($this->emailAddress->addresses)) {
-            $this->retrieve();
-        }
+        $this->retrieve();
```

Comment:
In case, if it has no any email address, unnecessary to retrieve hole bean for an email address ID in return value at the end.

Reply:
When you retrieve the bean it will get the email addresses. I noticed that emails module requires the retrieve.

Reply:
Still don't need to retrieve the all of the Basic bean even if it's already retrieved. Check the ID first if it's more make sense for you.